plugins/spansh_core.py

```diff
@@ -0,0 +1,195 @@
+"""
+spansh_core.py - Spansh URL provider.
+
+Copyright (c) EDCD, All Rights Reserved
+Licensed under the GNU General Public License.
+See LICENSE file.
+
+This is an EDMC 'core' plugin.
+All EDMC plugins are *dynamically* loaded at run-time.
+
+We build for Windows using `py2exe`.
+`py2exe` can't possibly know about anything in the dynamically loaded core plugins.
+
+Thus, you **MUST** check if any imports you add in this file are only
+referenced in this file (or only in any other core plugin), and if so...
+
+    YOU MUST ENSURE THAT PERTINENT ADJUSTMENTS ARE MADE IN
+    `build.py` TO ENSURE THE FILES ARE ACTUALLY PRESENT
+    IN AN END-USER INSTALLATION ON WINDOWS.
+"""
+from __future__ import annotations
+
+import tkinter as tk
+from typing import Any, cast
+import requests
+from companion import CAPIData
+from config import appname, config
+from EDMCLogging import get_main_logger
+
+logger = get_main_logger()
+
+
+class This:
+    """Holds module globals."""
+
+    def __init__(self):
+        self.parent: tk.Tk
+        self.shutting_down = False  # Plugin is shutting down.
+        self.system_link: tk.Widget = None  # type: ignore
+        self.system_name: str | None = None  # type: ignore
+        self.system_address: str | None = None  # type: ignore
+        self.system_population: int | None = None
+        self.station_link: tk.Widget = None  # type: ignore
```

Comment:
If assigning to None, should this be `tk.Widget | None`?

Reply:
``self.station_link`` should only be ``None`` while the main window is being created, once ``plugin_app()`` gets called ``self.station_link`` becomes a ``tk.Widget`` and shouldn't become ``None`` again.